src/midlewares/isUserInRoom.js

```
@@ -0,0 +1,19 @@
+import { roomService } from '../services/room.service.js';
+
+export const isUserInRoom = async (req, res, next) => {
+  const user = req.user;
+  const { roomId } = req.params;
+  const usersRooms = await roomService.findRoomsByUserId(user.id);
+
+  const room = usersRooms.find((r) => r.id === parseInt(roomId, 10));
```

`roomId` is taken from req.params and immediately passed into parseInt later. Validate that `roomId` exists and is a valid integer first. For example: `const parsedRoomId = parseInt(roomId, 10); if (!roomId || Number.isNaN(parsedRoomId)) return res.status(400).json({ error: 'Invalid roomId' })`. This prevents silent failures and makes response codes clearer for client errors.